zio-http/src/main/scala/zhttp/service/Handler.scala

```diff
@@ -156,22 +95,18 @@ private[zhttp] final case class Handler[R](
         if (self.isWebSocket(res)) {
           self.upgradeToWebSocket(ctx, jReq, res)
         } else {
-          // Write the initial line and the header.
-          unsafeWriteAndFlushAnyResponse(res)
-          res.data match {
-            case HttpData.BinaryStream(stream) => unsafeRunZIO(writeStreamContent(stream) *> Task(releaseRequest(jReq)))
-            case HttpData.File(file)           =>
-              unsafeWriteFileContent(file)
-            case _                             => releaseRequest(jReq)
-          }
+          ctx.fireChannelRead(res)
+          releaseRequest(jReq)
         }
-      case HExit.Failure(e)   =>
-        unsafeWriteAndFlushErrorResponse(e)
+
+      case HExit.Failure(e) =>
+        ctx.fireExceptionCaught(e)
```

Comment:
```suggestion
        ctx.fireChannelRead(e)
```

It's not an error in this context. We have an 500 status code response which is essentially a success value.

Reply:
right. updated code